docs/source/deep-learning/tensorflow/guide/index.rst

```diff
@@ -0,0 +1,269 @@
+You can then use the callback in ``model.fit()``. Here is a simple example for logging the training metrics
+in log scale:
+
+.. code-block::
+
+    from tensorflow import keras
+    import math
+    import mlflow
+
+    class MLflowCallback(keras.callbacks.Callback):
+        def on_epoch_end(self, epoch, logs=None):
+            logs = logs or {}
+            for k, v in logs.items():
+                mlflow.log_metric(f"log_{k}", math.log(v), step=epoch)
+
+Briefly ``logs`` object will contain your training metrics (``loss`` and ``metrics`` as defined in ``model.compile()``),
+during training and validation metrics during validation. For full documentation of the Keras callback API, please
+read `keras.callbacks.Callback <https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/Callback>`_.
+
+Saving Your Tensorflow Model to MLflow
+--------------------------------------
+
+If you have turned on the autologging, your Tensorflow model will be automatically saved after the training
+is done, or you can explicitly save you model to MLflow by calling :py:func:`mlflow.tensorflow.log_model()`.
+After saving, you can load back the model using :py:func:`mlflow.tensorflow.load_model()`. The loaded model
+can be used for inference by calling ``predict()`` method.
+
+.. code-block:: python
+
+    import mlflow
+    import tensorflow as tf
+    from tensorflow import keras
+
+    model = keras.Sequential(
+        [
+            keras.Input([28, 28, 3]),
+            keras.layers.Conv2D(8, 2),
+            keras.layers.MaxPool2D(2),
+            keras.layers.Flatten(),
+            keras.layers.Dense(2),
+            keras.layers.Softmax(),
+        ]
+    )
+
+    save_path = "model"
+    with mlflow.start_run() as run:
+        mlflow.tensorflow.log_model(model, "model")
+
+    # Load back the model.
+    loaded_model = mlflow.tensorflow.load_model(f"runs:/{run.info.run_id}/{save_path}")
+
+    print(loaded_model.predict(tf.random.uniform([1, 28, 28, 3])))
+
+
+Diving into Saving
+^^^^^^^^^^^^^^^^^^
+
+Under the hood of saving, we are converting the Tensorflow model into a pyfunc model, which is a generic
+type of model in MLflow. The pyfunc model is saved to MLflow. You don't need to learn the basics of pyfunc
+model to use Tensorflow flavor, but if you are interested, please refer to `MLflow pyfunc model <https://mlflow.org/docs/latest/models.html#how-to-load-and-score-python-function-models>`_.
+
+Saving Format
+~~~~~~~~~~~~~
+
+By default MLflow saves your Tensorflow model in the format of TF saved model (compiled graph), which is
+suitable for deployment across platforms. You can also save your model in other formats, i.e., ``h5`` and
+``keras`` by setting the ``keras_model_kwargs`` parameter in :py:func:`mlflow.tensorflow.log_model()`. For
+example. if you want to save your model in ``h5`` format, which only saves model weights instead of the
+compiled graph, you can run:
```

Comment:
```suggestion
example, if you want to save your model in ``h5`` format (which only saves model weights instead of the
compiled graph) you can run:
```